dhall/src/Dhall/Core.hs

```diff
@@ -1823,6 +1837,7 @@ isNormalized e0 = loop (denote e0)
           App NaturalEven (NaturalLit _) -> False
           App NaturalOdd (NaturalLit _) -> False
           App NaturalShow (NaturalLit _) -> False
+          App (App NaturalSubtract (NaturalLit _)) (NaturalLit _) -> False
```

Comment:
I believe you'll also need to handle `subtract x 0` and `subtract 0 x` here.

Reply:
Hmm, I can't get a test failure. Let me try in the repl.

Reply:
Yea, you're right - added those rules.

Reply:
Cheers!